mqtt/mqtt-broker/src/broker.rs

```diff
@@ -1987,4 +2315,82 @@ pub(crate) mod tests {
 
         Ok((client_id, rx))
     }
+
+    async fn disconnect_client(client_id: &str, broker_handle: &mut BrokerHandle) {
+        let event = ClientEvent::Disconnect(proto::Disconnect {});
+
+        broker_handle
+            .send(Message::Client(client_id.into(), event))
+            .await
+            .unwrap();
+    }
+
+    async fn send_subscribe(
+        handle: &mut BrokerHandle,
+        rx: &mut Receiver<Message>,
+        client_id: ClientId,
+        topics: &[&str],
```

Comment:
I noticed that you create a static array in the test case and then pass it as a reference, while you can move and consume in some mehtods. It will reduce manipulations with pointers.

Reply:
Its a test, and passing by reference let me avoid making the functions take generics

Reply:
Generics? I was talking about creating strings and move them instead of borrowing. Maybe I didn't understand your question

Reply:
ah, I thought you meant pass the array by value, not the strings in the array. Since its a test, I'd rather have it cleaner than more efficient, and its much cleaner to pass in `&["foo", "bar"]` than `&["foo".to_owned(), "bar".to_owned()]`